libminifi/test/integration/C2MetricsTest.cpp

```diff
@@ -126,28 +130,60 @@ class MetricsHandler: public HeartbeatHandler {
     }
   }
 
+  static bool processorMetricsAreValid(const auto& processor) {
+    return processor["bytesRead"].GetInt() >= 0 &&
+      processor["bytesWritten"].GetInt() >= 0 &&
+      processor["flowFilesIn"].GetInt() >= 0 &&
+      processor["flowFilesOut"].GetInt() >= 0 &&
+      processor["bytesIn"].GetInt() >= 0 &&
+      processor["bytesOut"].GetInt() >= 0 &&
+      processor["invocations"].GetInt() >= 0 &&
+      processor["processingNanos"].GetInt() >= 0 &&
+      processor["activeThreadCount"].GetInt() == -1 &&
+      processor["terminatedThreadCount"].GetInt() == -1 &&
+      processor["running"].GetBool();
+  }
+
   static bool verifyRuntimeMetrics(const rapidjson::Value& runtime_metrics) {
     return runtime_metrics.HasMember("deviceInfo") &&
       runtime_metrics.HasMember("flowInfo") &&
       runtime_metrics["flowInfo"].HasMember("versionedFlowSnapshotURI") &&
       runtime_metrics["flowInfo"].HasMember("queues") &&
-      runtime_metrics["flowInfo"].HasMember("components") &&
       runtime_metrics["flowInfo"]["queues"].HasMember("2438e3c8-015a-1000-79ca-83af40ec1997") &&
-      runtime_metrics["flowInfo"]["components"].HasMember("FlowController") &&
-      runtime_metrics["flowInfo"]["components"].HasMember("GetTCP") &&
-      runtime_metrics["flowInfo"]["components"].HasMember("LogAttribute");
+      runtime_metrics["flowInfo"].HasMember("processorStatuses") &&
+      [&]() {
+        if (runtime_metrics["flowInfo"]["processorStatuses"].GetArray().Size() != 2) {
+          return false;
+        }
+        const auto processor_statuses = runtime_metrics["flowInfo"]["processorStatuses"].GetArray();
```

Comment:
```suggestion
        const auto processor_statuses = runtime_metrics["flowInfo"]["processorStatuses"].GetArray();
        if (processor_statuses.Size() != 2) {
          return false;
        }
```

Reply:
Updated in https://github.com/apache/nifi-minifi-cpp/pull/1909/commits/36f3262ba5901b9f59828db052cb9ac1a5bdef2e